pgvectorscale/src/access_method/distance.rs

```diff
@@ -102,6 +105,9 @@ pub fn distance_cosine(a: &[f32], b: &[f32]) -> f32 {
         return super::distance_x86::distance_cosine_x86_avx2(a, b);
     }
 
+    #[cfg(all(any(target_arch = "arm", target_arch = "aarch64"), target_feature = "neon"))]
```

Comment:
```suggestion
    #[cfg(all(any(target_arch = "arm", target_arch = "aarch64")))]
```

Neon being missing should throw a compile error on arm right